runtime/parser2/lexer/lexer.go

```diff
@@ -270,10 +270,30 @@ func (l *lexer) endPos() position {
 }
 
 func (l *lexer) emitType(ty TokenType) {
+	if l.memoryGauge != nil {
+		usage := l.typeMemoryUsage(ty)
+
+		// Don't use `common.MemoryUsae()` to avoid redundant `nil` check.
+		err := l.memoryGauge.MeterMemory(usage)
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
I'm not sure whether we need to meter these types of tokens as they are not part of the AST (discarded). But at the same time, they are available in the token stream - so it's sort of in-memory

Reply:
Yes, I think we need to meter them, as they are allocations resulting from user interaction (code submission) and kept in memory, even if just temporarily